src/autocomplete.js

```diff
@@ -672,9 +672,17 @@ yourlabs.Autocomplete.prototype.makeXhr = function() {
 
 // Callback for the ajax response.
 yourlabs.Autocomplete.prototype.fetchComplete = function(jqXHR, textStatus) {
+    if (this.xhr === jqXHR) {
+        // Current request finished.
+        this.xhr = false;
+    } else {
+        // Ignore response from earlier request.
+        return;
```

Comment:
A future improvement here would be to not ignore the response, since it's a valid response. If every XHR gets an increasing ID attached to it and the autocomplete keeps track of the last displayed ID, then results from earlier could be safely displayed to the user as they type. The `xhr-pending` class would remain on the input until the current request finished, making it clear that the results are not final.